### PR TITLE
Allow to set default securityContextRepository for each authenticatio…

### PR DESCRIPTION
**Pull request by eddumelendez**
…n mechanisms

Fixes gh-7249

### CODE REVIEW

**Comment by eddumelendez** [comment]
@jzheaux how should `securityContextRepositoryWebFilter()` resolve the `securityContextRepository`? Since the new approach is:

1. Use `securityContextRepository` in the authentication mechanism
2. Use `securityContextRepository` in the global value
3. Use the default one

My current PR is setting the global `securityContextRepository` in the `build()` method by each authentication mechanism which I assume is wrong but didn't notice until running integration tests with configs with multiple authentication mechanism. Should the global value remain with an instance of `WebSessionServerSecurityContextRepository`?

Note: there have been some changes in other tests since http basic and auth2 login never read the `securityContextRepository` and there were some of them changing that value but it was ignored.

Thanks in advance

**Comment by jzheaux** [comment]
@eddumelendez Good observations.

>  Should the global value remain with an instance of WebSessionServerSecurityContextRepository?

No, good catch, I think that this should be set to `null`, and then also change `securityContextRepositoryWebFilter()` to use `WebSessionServerSecurityContextRepository` by default instead of returning `null`.

File: config/src/test/java/org/springframework/security/config/annotation/web/reactive/EnableWebFluxSecurityTests.java [hunk 1]
```diff
@@ -153,6 +153,7 @@ public void defaultPopulatesReactorContext() {
 		client
 			.get()
 			.uri("/")
+			.headers(headers -> headers.setBasicAuth("user", "password"))
```
Comment (jzheaux) [inline config/src/test/java/org/springframework/security/config/annotation/web/reactive/EnableWebFluxSecurityTests.java]: Why was this added? It seems like an odd side-effect of the change.

Comment (eddumelendez) [reply config/src/test/java/org/springframework/security/config/annotation/web/reactive/EnableWebFluxSecurityTests.java]: yes, you are right

**Comment by eddumelendez** [comment]
@jzheaux thanks again for all your feedback. I have submitted a new commit

**Comment by jzheaux** [comment]
Great, @eddumelendez, go ahead and squash your commits in preparation for merging.

**Comment by eddumelendez** [comment]
@jzheaux done :)
